fritzconnection/lib/fritzwlan.py

```diff
@@ -48,18 +82,26 @@ def get_beacon_security(instance, security):
         info = instance.get_info()
         # check for the POSSIBLE_BEACON_TYPES_KEY argument
         # as older models may not provide it:
+        beacontype = info["NewBeaconType"]
         if POSSIBLE_BEACON_TYPES_KEY in info:
             beacontypes = set(info[POSSIBLE_BEACON_TYPES_KEY].split(","))
-            beacontypes -= set(('None', 'OWETrans'))
-            beacontype = info["NewBeaconType"]
+            beacontypes -= set(("None", "OWETrans"))
             if beacontype in beacontypes:
-                security = WPA_SECURITY
-    return security
-
-
-def get_wifi_qr_code(instance, kind='svg',
-                     security=None, hidden=False,
-                     scale=4):
+                security = Security.WPA_SECURITY
+        elif beacontype in {security.value for security in Security}:
```

Comment:
This will get only executed, if the devices supports a beacon type, but does not report about the supported beacon types. In this case you check whether security should be set to "no_pass", "WPA", or "WPA2".